cadc-wcs/src/main/java/ca/nrc/cadc/wcs/WCSLib.java

```diff
@@ -72,53 +72,77 @@
 import ca.nrc.cadc.wcs.Transform.Result;
 import ca.nrc.cadc.wcs.exceptions.WCSLibInitializationException;
 import ca.nrc.cadc.wcs.exceptions.WCSLibRuntimeException;
+
 import java.util.Map;
 import java.util.TreeMap;
-import java.util.UUID;
 import org.apache.log4j.Logger;
 
 /**
  * This class contains the native method definitions to the WCSLIB 4.2 C routines
- * wcss2p() and wcsp2s(), and provides package level access to methods implementing 
+ * wcss2p() and wcsp2s(), and provides package level access to methods implementing
  * the native definitions.
- * 
+ *
  * @author jburke
  *
  */
 final class WCSLib
 {
-    private static final Logger log = Logger.getLogger(WCSLib.class);
-    
-    private static final UUID uuid = UUID.randomUUID();
-    
+    private static final Logger LOGGER = Logger.getLogger(WCSLib.class);
+    private static final String JNI_SO_PREFIX = "libwcsLibJNI";
+    private static final String[] JNI_FILES = new String[] {
+            JNI_SO_PREFIX + ".5.so",
+            JNI_SO_PREFIX + ".6.so",
+            JNI_SO_PREFIX + ".7.so",
```

Comment:
I would order these from latest backwards so that someone with the latest wcslib would not see the error messages... might even encourage people to update.

And you could remove the `.so` here and then not need to strip it off in the loop below.